thali/thalicryptomanager.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+var crypto = require('crypto');
+var fs = require('fs');
+
+var pkcs12FileName = '/pkcs12.pfx';
+var password = 'password';
+var certname = 'certname';
+var country = 'country';
+var organization = 'organization';
+var macName = 'SHA256';
+var hashSizeInBytes = 16;
+
+module.exports = {
+
+/**
+* Creates a PKCS12 content, saves it to a file, extracts the public
+* key and returns it's SHA256 hash value.
+*/  
+  getPublicKeyHash: function (callback) {
+    // get the path where the PKCS12 file is saved
+    Mobile.GetDocumentsPath(function (err, fileLocation) {
+      if (err) {
+        console.error("GetDocumentsPath err: ", err);
+        callback(null);
```

Comment:
We should follow the usual node.js convention of returning an error in the first argument and success information in the second argument. So we should be returning an error string in the first argument here not null.